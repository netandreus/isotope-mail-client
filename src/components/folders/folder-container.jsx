import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {translate} from 'react-i18next';
import Spinner from '../spinner/spinner';
import IconButton from '../buttons/icon-button';
import FolderList from './folder-list';
import FolderCreateDialog from './folder-create-dialog';
import FolderRenameDialog from './folder-rename-dialog';
import {createFolder as createFolderAction} from '../../actions/application';
import styles from './folder-container.scss';
import mainCss from '../../styles/main.scss';

export const FolderContainer = ({t, activeRequests, folderTree, folders, createFolder}) => {
  const initialListLoading = activeRequests > 0 && folderTree.length === 0;
  return (
    <nav className={`${mainCss['mdc-list']}`}>
      <Spinner visible={initialListLoading}
        canvasClassName={styles.spinnerCanvas} />
      <FolderList folderTree={folderTree} folders={folders}/>
      {!initialListLoading &&
      (<div isotip={t('sideBar.newFolder')} isotip-position='top' isotip-size='small'>
        <IconButton
          className={`${styles.addButton}`} onClick={createFolder}>
          add_circle
        </IconButton>
      </div>)}
      <FolderCreateDialog />
      <FolderRenameDialog />
    </nav>
  );
};

FolderContainer.propTypes = {
  activeRequests: PropTypes.number,
  folderTree: PropTypes.array,
  folders: PropTypes.object,
  createFolder: PropTypes.func
};

const mapStateToProps = state => ({
  activeRequests: state.folders.activeRequests,
  folderTree: state.folders.items,
  folders: state.folders.explodedItems
});

const mapDispatchToProps = dispatch => ({
  createFolder: () => dispatch(createFolderAction(''))
});

export default connect(mapStateToProps, mapDispatchToProps)(translate()(FolderContainer));
