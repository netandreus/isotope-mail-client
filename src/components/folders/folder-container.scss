@import '../../styles/partials/variables';

.spinnerCanvas {
  width: 24px;
}

.addButton {
  display: block;
  margin: 0 auto;
  color: $mdc-theme-primary;
}
