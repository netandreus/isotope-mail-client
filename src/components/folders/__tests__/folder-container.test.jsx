import React from 'react';
import {shallow} from 'enzyme';
import {FolderContainer} from '../folder-container';

describe('FolderContainer component test suite', () => {
  test('Snapshot render, should render FolderContainer', () => {
    // Given
    const props = {
      t: key => key,
      activeRequests: 0,
      folders: {},
      folderTree: [{folders: {}, folderTree: []}],
      selectedFolder: {folderId: '1337-3'}
    };

    // When
    const folderContainer = shallow(<FolderContainer {...props} />);

    // Then
    expect(folderContainer).toMatchSnapshot();
  });
});
