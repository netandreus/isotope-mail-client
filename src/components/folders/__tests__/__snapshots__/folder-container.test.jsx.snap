// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderContainer component test suite Snapshot render, should render FolderContainer 1`] = `
<nav
  className="mdc-list"
>
  <Spinner
    canvasClassName="spinnerCanvas"
    className=""
    pathClassName=""
    visible={false}
  />
  <Connect(FolderListClass)
    folderTree={
      Array [
        Object {
          "folderTree": Array [],
          "folders": Object {},
        },
      ]
    }
    folders={Object {}}
  />
  <div
    isotip="sideBar.newFolder"
    isotip-position="top"
    isotip-size="small"
  >
    <IconButton
      className="addButton"
      disabled={false}
      onClick={[Function]}
    >
      add_circle
    </IconButton>
  </div>
  <Connect(Translate(FolderCreateDialog)) />
  <Connect(Translate(FolderRenameDialog)) />
</nav>
`;
